server/requestorserver/auth.go

```diff
@@ -243,13 +243,27 @@ func jwtAutheticateRevocation(
 	if !jwtApplies(headers, body, signatureAlg) {
 		return false, nil, "", nil
 	}
-	s := &irma.RevocationJwt{}
-	if _, err := jwt.ParseWithClaims(string(body), s, jwtKeyExtractor(keys)); err != nil {
-		return false, nil, "", server.RemoteError(server.ErrorInvalidRequest, err.Error())
+
+	claims := &jwt.StandardClaims{}
+	revocationJwt := string(body)
+	if _, err := jwt.ParseWithClaims(revocationJwt, claims, jwtKeyExtractor(keys)); err != nil {
+		return true, nil, "", server.RemoteError(server.ErrorInvalidRequest, err.Error())
 	}
-	if time.Unix(time.Time(s.IssuedAt).Unix(), 0).Add(time.Duration(maxRequestAge) * time.Second).Before(time.Now()) {
+	if time.Unix(claims.IssuedAt, 0).Add(time.Duration(maxRequestAge) * time.Second).Before(time.Now()) {
 		return true, nil, "", server.RemoteError(server.ErrorUnauthorized, "jwt too old")
 	}
+	if !claims.VerifyIssuedAt(time.Now().Unix(), true) {
+		return true, nil, "", server.RemoteError(server.ErrorUnauthorized, "jwt not yet valid")
+	}
+
+	// Read JWT contents
+	s := &irma.RevocationJwt{}
+	if _, _, err := new(jwt.Parser).ParseUnverified(revocationJwt, s); err != nil {
+		return true, nil, "", server.RemoteError(server.ErrorInvalidRequest, err.Error())
+	}
+	if err := s.Request.Validate(); err != nil {
+		return true, nil, "", server.RemoteError(server.ErrorInvalidRequest, "Invalid JWT body")
+	}
```

Comment:
There is quite some code duplication with the `jwtAuthenticate` function above. This is one of the reasons the revocation JWT had a bug and the requestor JWT didn't. Do you have some ideas to resolve this?

Reply:
True, I have introduced an additional function `jwtValidateClaims`.
I am not happy about the dependency on `jwt.StandardClaims`, but that seems out of scope for this proposal.

Reply:
Yeah I agree that it looks a bit weird due to this. But for now that's okay I think.